docs/pages/versions/unversioned/workflow/testing-with-jest.md

```diff
@@ -0,0 +1,171 @@
+- [Cli Options](https://jestjs.io/docs/en/cli)
+
+## Code Coverage Reports
+
+Running tests are cool and all, but wouldn't you like to see your Expo projects total code coverage?! Maybe even see it in html format?! This section is for you!
+
+Let's head back to `package.json` and add the following:
+
+```js
+"jest": {
+  ...
+  "collectCoverage": true,
+  "collectCoverageFrom": [
+    "**/*.{js,jsx}",
+    "!**/coverage/**",
+    "!**/node_modules/**",
+    "!**/babel.config.js",
+    "!**/jest.setup.js"
+  ],
+  "verbose": true
+}
+```
+
+The above additions let's Jest know to collect coverage of all ***.js & .jsx*** file types and not within ***/coverage***, ***/node_modules/*** and our 2 project config files (add/remove more exclusions to this list to match your Expo app needs). With [verbose](https://jestjs.io/docs/en/cli#verbose) set as true.
+
+Now run the test again, you should see **/coverage/** in your app directory! Find the `index.html` file within and double click to open it up in a browser. Not only do we have reporting in our cli, we also have an html version of our code coverage, pretty cool stuff!
+
+> **Standards Note**: You can do what you want, but usually we wouldn't upload this html reporting to git; so add `coverage/\*\*/*` as a line in `.gitignore` to prevent this directory from being tracked.
+
+## Structure your Tests
+
+As promised, let's talk about how to set up the tests, right now we have a single `.test.js` in the root of our Expo project. This can get messy quickly, the easiest way is to create a `__tests__` directory (anywhere you'd like) and put all the tests there. For example see below:
+
+```bash
+__tests__/
+├─ components/
+│  └─ button.test.js
+├─ navigation/
+│  └─ mainstack.test.js
+└─ screens/
+   └─ home.test.js
+src/
+├─ components/
+│  └─ button.js
+├─ navigation/
+│  └─ mainstack.js
+└─ screens/
+   └─ home.js
+```
+
+This **could** cause a lot of long import paths though (`../../src/components/button`), so another option would be to have multiple `__tests__` directories:
+
+```bash
+src/
+├─ components/
+├─ __tests__/
+│  │  └─ button.test.js
+│  └─ button.js
+...
+```
+
+So if we move `__tests__` within `components`, the *button.test.js* import of `<Button />` would now be: `../button`, that's a lot better!
+
+Another option for test/file structure:
+
+```bash
+src/
+├─ components/
+│  ├─ button.js
+│  ├─ button.style.js
+│  └─ button.test.js
+...
+```
+
+At the end of the day, it's all preference and up to you on how you'd like to structure things, but we wanted to share a few options to consider.
+
+## Jest Flows
+
+This is optional, but wanted to talk about different jest test flows. Currently we run the test and that's it, but take a look at the [Cli Options](https://jestjs.io/docs/en/cli). Below are a few script setups to try:
+
+```js
+"scripts": {
+  ...
+  // active development of tests, watch files for changes and re-runs all tests
+  "test": "jest --watchAll --coverage=false",
```

Comment:
why `watchAll` instead of just `watch`? All files are still watched no matter what, it's just "should jest run all tests, or just those that have changed since the last commit"

Reply:
might be worth to add a `--changedSince=origin/master`?